backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/SqlSchemaMetadata.java

```diff
@@ -229,4 +232,26 @@ private static SqlSchemaMetadata renameTableTransaction(
     sm.tables.put(newName, tm);
     return sm;
   }
+
+  public List<String> getIneritedRolesForUser(String user) {
+    if (user == null) return new ArrayList<>();
+    // add cache because this function is called often
+    if (rolesCache != null) return rolesCache;
```

Comment:
exposed private / state-full value of roles list , suggest te return immutable copy

Reply:
good point, will add before merging